core/lib/spree/core/product_duplicator.rb

```diff
@@ -72,7 +72,7 @@ def reset_properties
     end
 
     def sku_generator(sku)
-      "COPY OF #{Variant.where('sku like ?', "%#{sku}").order(:created_at).last.sku}"
+      "COPY OF #{Variant.with_deleted.where('sku like ?', "%#{sku}").order(:created_at).last.sku}"
```

Comment:
@tanmay3011 I don't think it will raise an exception. As the sku value is coming from the sku of the product being cloned.
But there are some other issues related to this ``sku_genearator`` logic.

**First Issue**
Product 1 sku is TEST123
Product 2 sku is MY-TEST123

Now if we try to clone product 1 and if it is created before the second product then the sku of clone product would be ``Copy of MY-TEST123`` instead of ``Copy of TEST123``

**Second Issue** 
If sku of a product is empty and it is get cloned then sku of cloned product would be ``Copy of`` which obviously looks incomplete.

@damianlegawiec @tanmay3011 Guys, what's your thought on this?
I can raise a PR with a possible fix for this if you guys think it's needed to be fixed :)